scripts/lib/LinkChecker.ts

```diff
@@ -117,6 +117,52 @@ export class Link {
     );
   }
 
+  /**
+   * Returns a string with a suggested replacement for a broken link
+   */
+  didYouMean(existingFiles: File[], originFile: string): string {
+    const levenshtein = require("fast-levenshtein");
+
+    // Find a new valid link
+    let min_score = Number.MAX_SAFE_INTEGER;
+    let suggestion_path: String = "";
+    let suggestion_path_anchors: String[] = [];
+
+    const possiblePaths = this.possibleFilePaths(originFile);
+    const pathNoExtension = possiblePaths[0].replace(/\.[^\/.]+$/, "");
+
+    existingFiles.forEach((file) => {
+      let score = levenshtein.get(pathNoExtension, file.path);
+      if (score < min_score) {
+        min_score = score;
+        suggestion_path = file.path;
+        suggestion_path_anchors = file.anchors;
+      }
+    });
+
+    if (this.anchor == "") {
+      return `❓ Did you mean '${suggestion_path
+        .replace(/\.[^\/.]+$/, "")
+        .replace(/^docs/, "")}'?`;
+    }
+
+    // Find a new valid anchor
+    min_score = Number.MAX_SAFE_INTEGER;
```

Comment:
Does it make sense to use a new variable for clarity?

Reply:
Yes, maybe it's more readable, done!